krabs/krabs/filtering/comparers.hpp

```diff
@@ -48,10 +45,9 @@ namespace krabs { namespace predicates {
             template <typename Iter1, typename Iter2>
             bool operator()(Iter1 begin1, Iter1 end1, Iter2 begin2, Iter2 end2) const
             {
-                auto r1 = boost::make_iterator_range(begin1, end1);
-                auto r2 = boost::make_iterator_range(begin2, end2);
-
-                return boost::contains(r1, r2, Comparer());
+                // semantics of boost::contains() require that empty range is always contained
+                return std::distance(begin2, end2) == 0 ||
```

Comment:
No need for the special handling for when dist2 is 0 because since C++11, the standard requires std::search() to return begin1 instead of end2 in this case.

Reply:
Agreed. Problem here is when both the target sequence and the test sequence are empty. True to the standard, std::search() still returns begin1, but in this case begin1 == end1, so ultimately the function erroneously returns false. The check against an empty test sequence is the most expressive way I can see to handle this because it is the only case in which an empty target sequence contains any test sequence, so we can short-circuit the std::search() test.